server/service/src/invoice/outbound_return/update_lines/validate.rs

```diff
@@ -0,0 +1,29 @@
+use repository::{InvoiceRowType, StorageConnection};
+
+use crate::invoice::{
+    check_invoice_exists, check_invoice_is_editable, check_invoice_type, check_store,
+};
+
+use super::UpdateOutboundReturnLinesError;
+
+pub fn validate(
```

Comment:
These checks are actually done for each line in the stockOutLine insert/update/delete validations... not sure how much value it adds to do it at the outer layer, except maybe it's nice to have a direct error, rather than these bubbling back from one of the line mutations??

Reply:
Not seeing what you mean re running for each line. 
It's this isn't it?
```
    let outbound_return = ctx
        .connection
        .transaction_sync(|connection| {
            validate(connection, &ctx.store_id, &input.outbound_return_id)?;
```

Reply:
Oh, just that the invoice exists/is_editable etc. checks are also run in the insert/update_stock_out_line validation methods, so we're kind of doubling up on those checks. However I think it is best to keep them in both 👍